test/conformance/api/volumes_test.go

```diff
@@ -72,18 +73,97 @@ func TestConfigMapVolume(t *testing.T) {
 			LocalObjectReference: corev1.LocalObjectReference{
 				Name: configMap.Name,
 			},
+			Optional: ptr.Bool(false),
 		},
-	},
-	)
+	})
 
-	// Setup initial Service
-	if _, err := test.CreateRunLatestServiceReady(t, clients, &names, &test.Options{}, withVolume); err != nil {
+	withBadVolume := WithVolume("blah", "/does/not/matter", corev1.VolumeSource{
```

Comment:
nit: `withOptionalBadVolume` would be a bit more clear that this volume is testing the Optional condition and that passing a bad volume should result in a working service

Reply:
RFAL